File: frontend/src/interfaces/Bid.ts
import { AuctionInterface } from "./Auction";
import { BiddersInterface } from "./IBidder";

export interface BidsInterface {
    ID?: number;
    BidAmount?: number;
    AuctionDetail?: AuctionInterface;
    AuctionDetailID?: number;
    Bidder?: BiddersInterface;
    BidderID?: number;
}

File: backend/entity/bid.go
package entity

import (

	"gorm.io/gorm"
)

type Bid struct {
	gorm.Model

	BidAmount int `valid:"required~Bid is required, customBidMin~Bid must be a non-negative integer"`
	
	AuctionDetailID     uint    `valid:"required~Auction Detail is required"`
	AuctionDetail       AuctionDetail  `gorm:"foreignKey:AuctionDetailID"` 

	BidderID     uint    `valid:"required~Auction Detail is required"`
	Bidder       Bidder  `gorm:"foreignKey:BidderID"`

}